mlrun/projects/project.py

```diff
@@ -3115,6 +3115,13 @@ def build_image(
 
             function = mlrun.new_function("mlrun--project--image--builder", kind="job")
 
+            if self.spec.source and not self.spec.load_source_on_run:
+                function.with_source_archive(
+                    source=self.spec.source,
+                    workdir=self.spec.workdir,
```

Comment:
workdir is not part of the build job but it is set during runtime (before starting the run)

Reply:
huh you are right :( that's my fault.. we should move it to the build spec

Reply:
It's also not part of `project.set_source`, and it does make sense. I think for the project use-case, it's ok to have it as a parameter for `build_image`, which is what I did.